src/renderer/components/deposit/withdraw/Withdraw.tsx

```diff
@@ -1,65 +1,282 @@
 import { getWithdrawAmounts } from './Withdraw.helper'
 import * as Styled from './Withdraw.styles'
 
 type Props = {
-  depositAsset: Asset
+  /** Type to withdraw (`sym` or `asym`) */
+  type: WithdrawType
+  /** Asset to withdraw */
+  asset: Asset
+  /** Rune price */
   runePrice: BigNumber
+  /** Asset based `PoolData` */
+  assetPoolData: PoolData
+  /** Asset price */
   assetPrice: BigNumber
-  selectedCurrencyAsset: Asset
+  /** `PoolData` of assets chain asset */
+  chainAssetPoolData: PoolData
+  /** Wallet balance of Rune */
+  runeBalance: O.Option<BaseAmount>
+  /** Selected price asset */
+  selectedPriceAsset: Asset
+  /** Callback to send withdraw tx */
   onWithdraw: (percent: number) => void
-  updateFees: () => void
+  /** Callback to reload fees */
+  reloadFees: () => void
+  /** Share of Rune */
   runeShare: BaseAmount
+  /** Share of selected Asset */
   assetShare: BaseAmount
+  /** Flag whether form has to be disabled or not */
   disabled?: boolean
-  fee: WithdrawFeeRD
+  /** Fees needed to withdraw */
+  fees: WithdrawFeesRD
 }
 
 export const Withdraw: React.FC<Props> = ({
+  type,
+  asset,
+  assetPoolData,
   onWithdraw,
-  depositAsset,
+  asset: depositAsset,
   runePrice,
   assetPrice,
-  selectedCurrencyAsset,
+  chainAssetPoolData,
+  runeBalance: oRuneBalance,
+  selectedPriceAsset,
   runeShare,
   assetShare,
-  disabled: disabledProp,
-  fee,
-  updateFees
+  disabled,
+  fees,
+  reloadFees: updateFees
 }) => {
   const intl = useIntl()
-  const [withdrawPercent, setWithdrawPercent] = useState(disabledProp ? 0 : 50)
 
-  const disabled = useMemo(() => withdrawPercent === 0 || disabledProp, [withdrawPercent, disabledProp])
+  const isDesktopView = Grid.useBreakpoint()?.lg ?? false
+
+  const [withdrawPercent, setWithdrawPercent] = useState(disabled ? 0 : 50)
+
+  const _isAsym = useMemo(() => type === 'asym', [type])
 
   const withdrawAmounts = getWithdrawAmounts(runeShare, assetShare, withdrawPercent)
-  const renderFee = useMemo(
-    () => [
-      FP.pipe(
-        fee,
-        RD.map((fee) => ({
-          asset: BASE_CHAIN_ASSET,
-          amount: fee
-        })),
-        (feesRd) => <Fees fees={[feesRd]} />
+
+  const oFees: O.Option<WithdrawFees> = useMemo(() => FP.pipe(fees, RD.toOption), [fees])
+
+  const isThorMemoFeeError: boolean = useMemo(() => {
+    if (withdrawPercent <= 0) return false
+
+    return FP.pipe(
+      sequenceTOption(oFees, oRuneBalance),
+      O.fold(
+        // Missing (or loading) fees does not mean we can't sent something. No error then.
+        () => !O.isNone(oFees),
+        ([{ thorMemo }, balance]) => balance.amount().isLessThan(thorMemo.amount())
       )
-    ],
-    [fee]
+    )
+  }, [oFees, oRuneBalance, withdrawPercent])
+
+  const renderThorMemoFeeError = useMemo(() => {
+    const runeBalance = FP.pipe(
+      oRuneBalance,
+      O.getOrElse(() => ZERO_BASE_AMOUNT),
+      baseToAsset
+    )
```

Comment:
i know this is really small difference here but if we change order we can avoid calculations in case of `O.none`|

```ts
const runeBalance = FP.pipe(
      oRuneBalance,
      O.map(baseToAsset),
      O.getOrElse(() => ZERO_ASSET_AMOUNT),
    )
```

real nice feature of ADT is lazy functions. in more global sense it's better to use all advantages of this features